server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/aws/AwsS3ObjectStorage.java

```diff
@@ -137,12 +137,12 @@ private ContextBuilder contextBuilder() {
         private static final Duration FIRST_BACK_OFF = Duration.ofMillis(100);
         private static final Duration FOREVER = Duration.ofMillis(Long.MAX_VALUE);
 
-        private final AwsS3AuthConfiguration configuration;
-        private final ExecutorService executorService;
+        private final AmazonS3 s3Client;
+        private final TransferManager transferManager;
 
-        AwsS3BlobPutter(AwsS3AuthConfiguration configuration, ExecutorService executorService) {
-            this.configuration = configuration;
-            this.executorService = executorService;
+        AwsS3BlobPutter(AwsS3AuthConfiguration authConfiguration, ExecutorService executorService) {
+            this.s3Client = getS3Client(authConfiguration, getClientConfiguration());
+            this.transferManager = getTransferManager(s3Client, executorService);
```

Comment:
resources clean up is not easy to do as AwsS3BlobPutter is not the Bean managed by Guice

Reply:
I'm gonna create an issue about this in case you accept the current state of this component (no resources clean up supported).